packages/atclient/include/atclient/atclient.h

```diff
@@ -55,8 +55,8 @@ int atclient_start_secondary_connection(atclient *ctx, const char *secondaryhost
  * @param atsign the atsign the atkeys belong to
  * @return int 0 on success
  */
-int atclient_pkam_authenticate(atclient *ctx, atclient_connection *root_conn, const atclient_atkeys atkeys,
-                               const char *atsign, const size_t atsignlen);
+int atclient_pkam_authenticate(atclient *ctx, atclient_connection *root_conn, const atclient_atkeys *atkeys,
+                               const atclient_atsign *atsign);
```

Comment:
I would prefer if we kept this as a char*, I find managing atclient_atsign to be very irritating, since you have to deep copy to clone. Where as with char* you can use strcpy directly. This also reduces the surface for bugs, and pushes to be to use an idiomatic C approach, (instead of `atsign.without_at` [atclient_atsign], call `atsign + 1` [char*])

Reply:
https://github.com/atsign-foundation/at_c/pull/186/commits/431a4c2c3493dd1f9808a2ee61f42ba8104c06b4